src/framework/uicomponents/view/popupview.cpp

```diff
@@ -0,0 +1,262 @@
+}
+
+QPointF PopupView::globalPos() const
+{
+    return m_globalPos;
+}
+
+PopupView::ClosePolicy PopupView::closePolicy() const
+{
+    return m_closePolicy;
+}
+
+bool PopupView::isOpened() const
+{
+    return m_isOpened;
+}
+
+qreal PopupView::padding() const
+{
+    return m_padding;
+}
+
+void PopupView::setBackgroundItem(QQuickItem* backgroundItem)
+{
+    if (m_backgroundItem == backgroundItem)
+        return;
+
+    m_backgroundItem = backgroundItem;
+    emit backgroundItemChanged(m_backgroundItem);
+}
+
+void PopupView::setContentItem(QQuickItem* contentItem)
+{
+    if (m_contentItem == contentItem)
+        return;
+
+    m_contentItem = contentItem;
+    emit contentItemChanged(m_contentItem);
+}
+
+void PopupView::setClosePolicy(PopupView::ClosePolicy closePolicy)
+{
+    if (m_closePolicy == closePolicy)
+        return;
+
+    m_closePolicy = closePolicy;
+    emit closePolicyChanged(m_closePolicy);
+}
+
+void PopupView::setIsOpened(bool isOpened)
+{
+    if (m_isOpened == isOpened)
+        return;
+
+    m_isOpened = isOpened;
+    emit isOpenedChanged(m_isOpened);
+}
+
+void PopupView::setPadding(qreal padding)
+{
+    if (qFuzzyCompare(m_padding, padding))
+        return;
+
+    m_padding = padding;
+    emit paddingChanged(m_padding);
+}
+
+void PopupView::setPositionDisplacementX(qreal positionDisplacementX)
+{
+    if (qFuzzyCompare(m_positionDisplacementX, positionDisplacementX))
+        return;
+
+    m_positionDisplacementX = positionDisplacementX;
+    setX(positionDisplacementX);
+    emit positionDisplacementXChanged(m_positionDisplacementX);
+}
+
+void PopupView::setPositionDisplacementY(qreal positionDisplacementY)
+{
+    if (qFuzzyCompare(m_positionDisplacementY, positionDisplacementY))
+        return;
+
+    m_positionDisplacementY = positionDisplacementY;
+    setY(positionDisplacementY);
+    emit positionDisplacementYChanged(m_positionDisplacementY);
+}
+
+void PopupView::setGlobalPos(QPointF globalPos)
+{
+    if (m_globalPos == globalPos)
+        return;
+
+    m_globalPos = globalPos;
+    m_containerView->setPosition(globalPos.toPoint());
+    emit globalPosChanged(m_globalPos);
+}
+
+void PopupView::componentComplete()
+{
+    QQuickItem::componentComplete();
+
+    QQmlEngine* engine = qmlEngine(this);
+    QQmlContext* ctx = qmlContext(this);
+
+    setupContentComponent(engine);
+    setupContainerView(engine);
+    QQuickItem* obj = loadContentItem(ctx);
+
+    m_containerView->setContent(QUrl(), m_contentComponent, obj);
+    setSize(obj->size());
+}
+
+bool PopupView::eventFilter(QObject* watched, QEvent* event)
+{
+    if (QEvent::MouseButtonPress == event->type()) {
+        mousePressEvent(static_cast<QMouseEvent*>(event));
+    } else if (QEvent::MouseButtonRelease == event->type()) {
+        mouseReleaseEvent(static_cast<QMouseEvent*>(event));
+    }
+
+    return QObject::eventFilter(watched, event);
+}
+
+void PopupView::mousePressEvent(QMouseEvent* event)

```

Comment:
The implementation of this function duplicates mouseReleaseEvent

Reply:
not really, it handles 2 different modes of auto-close policy